cli/azd/main.go

```diff
@@ -58,6 +60,21 @@ func main() {
 	go fetchLatestVersion(latest)
 
 	cmdErr := cmd.NewRootCmd(false, nil).ExecuteContext(ctx)
+
+	if !isJsonOutput() && runcontext.IsFirstRun() && runcontext.IsRunningInCloudShell() {
```

Comment:
Interesting behavior to note here is that if the user only runs `azd` with `--json` output, we will never get telemetry on the usage. 

I wonder if we would favor displaying the message regardless of `--json` for the first run. `stderr` for `json` output is reserved primarily for console messages, wouldn't be a big deal if there's additional console messages displayed.

Reply:
This is a rare scenario. Most Cloud Shell interactions would happen without `--json` but it's not impossible. 

To consider for the future: In the structured output also include a message containing the telemetry notification. 

Since this is an unlikely scenario we'll keep things roughly as they are for now: only collect telemetry if a notice has been displayed to the user in the past.